File: Samples-Java/Samples/src/main/java/fragments/dataform/DataFormValidationModeFragment.java
package fragments.dataform;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.Spinner;

import com.telerik.android.sdk.R;
import com.telerik.widget.dataform.visualization.DataFormLinearLayoutManager;
import com.telerik.widget.dataform.visualization.RadDataForm;
import com.telerik.widget.dataform.visualization.core.CommitMode;
import com.telerik.widget.dataform.visualization.core.ValidationMode;
import com.telerik.widget.dataform.visualization.editors.adapters.EditorSpinnerAdapter;

import activities.ExampleFragment;

public class DataFormValidationModeFragment extends Fragment implements ExampleFragment, AdapterView.OnItemSelectedListener, View.OnClickListener {
    RadDataForm dataForm;
    Button validateButton;

    @Override
    public String title() {
        return "Validation Modes";
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        ViewGroup rootLayout = (ViewGroup) inflater.inflate(R.layout.fragment_dataform_validation_mode, null);

        Spinner validationModeSpinner = (Spinner) rootLayout.findViewById(R.id.data_form_validation_mode_spinner);
        validationModeSpinner.setOnItemSelectedListener(this);
        validationModeSpinner.setAdapter(new EditorSpinnerAdapter(this.getActivity(), R.layout.data_form_spinner_item, ValidationMode.values()));

        dataForm = new RadDataForm(this.getActivity());

        dataForm.setLayoutManager(new DataFormLinearLayoutManager(this.getActivity()));
        dataForm.setCommitMode(CommitMode.MANUAL);
        dataForm.setValidationMode(ValidationMode.IMMEDIATE);

        Person joe = new Person();
        joe.setName("Joe");
        joe.setMail("joe@mailservice.com");
        dataForm.setEntity(joe);

        validateButton = (Button)rootLayout.findViewById(R.id.data_form_validate_button);
        validateButton.setOnClickListener(this);

        rootLayout.addView(dataForm, 0);

        return rootLayout;
    }

    @Override
    public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
        dataForm.setValidationMode(ValidationMode.values()[position]);

        // Manual validation
        if(position == 2) {
            validateButton.setVisibility(View.VISIBLE);
        } else {
            validateButton.setVisibility(View.INVISIBLE);
        }
    }

    @Override
    public void onNothingSelected(AdapterView<?> parent) {

    }

    @Override
    public void onClick(View v) {
        dataForm.validateChanges();
    }
}

File: Samples-Xamarin/Fragments/Dataform/DataFormValidationModeFragment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Util;
using Android.Views;
using Android.Widget;

using Com.Telerik.Widget.Dataform.Visualization;
using Com.Telerik.Widget.Dataform.Visualization.Core;
using Com.Telerik.Widget.Dataform.Engine;
using Com.Telerik.Widget.Dataform.Visualization.Editors;
using Com.Telerik.Widget.Dataform.Visualization.Editors.Adapters;

namespace Samples
{
	public class DataFormValidationModeFragment : Android.Support.V4.App.Fragment, ExampleFragment, AdapterView.IOnItemSelectedListener, View.IOnClickListener
	{
		RadDataForm dataForm;
		Button validateButton;

		public String Title() {
			return "Validation Modes";
		}

		public override View OnCreateView (LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
		{
			ViewGroup rootLayout = (ViewGroup) inflater.Inflate(Resource.Layout.fragment_dataform_validation_mode, container, false);

			Spinner validationModeSpinner = (Spinner) rootLayout.FindViewById(Resource.Id.data_form_validation_mode_spinner);
			validationModeSpinner.OnItemSelectedListener = this;
			validationModeSpinner.Adapter = new ArrayAdapter(this.Activity, Android.Resource.Layout.SimpleSpinnerItem, new ValidationMode[]{ValidationMode.Immediate, ValidationMode.OnLostFocus, ValidationMode.Manual});

			dataForm = (RadDataForm)rootLayout.FindViewById(Resource.Id.data_form_validation_mode);

			dataForm.LayoutManager = new DataFormLinearLayoutManager(this.Activity);
			dataForm.CommitMode = CommitMode.Manual;
			dataForm.ValidationMode = ValidationMode.Immediate;

			Person joe = new Person();
			joe.Name = "Joe";
			joe.Mail = "joe@mailservice.com";
			dataForm.SetEntity (joe);

			validateButton = (Button)rootLayout.FindViewById(Resource.Id.data_form_validate_button);
			validateButton.SetOnClickListener(this);

			return rootLayout;
		}

		public void OnItemSelected(AdapterView parent, View view, int position, long id) {
			dataForm.ValidationMode = ValidationMode.Values()[position];

			// Manual validation
			if(position == 2) {
				validateButton.Visibility = ViewStates.Visible;
			} else {
				validateButton.Visibility = ViewStates.Invisible;
			}
		}

		public void OnNothingSelected(AdapterView parent) {

		}

		public void OnClick(View v) {
			dataForm.ValidateChanges();
		}
	}
}
